### PR TITLE
Profiles: New discover cards

### PR DESCRIPTION
**Pull request by christianbaroni**
## What changed (plus any additional context for devs)
Added new discover cards under the profiles flag

## PoW (screenshots / screen recordings)
Screen Shot 2022-06-16 at 2 04 38 AM

## Dev checklist for QA: what to test
- Verify new gas card, ENS search card, ENS profiles card, and DPI card work as expected
- Verify light mode and dark mode
- Verify these new cards on android
- Verify that by turning off the profiles flag, the old Discover screen still works as expected
- Verify that the "register a .eth name" updates when you have pending registrations

## Final checklist

- [x] Assigned individual reviewers?
- [x] Added labels?
- [ ] TODO - If you added new files, did you update the CODEOWNERS file?

### CODE REVIEW

File: src/components/discover-sheet/GasCard.tsx [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+      color = grey;
+    } else if (gwei < 40) {
+      color = green;
+    } else if (gwei < 100) {
+      color = blue;
+    } else if (gwei < 200) {
+      color = orange;
+    } else {
+      color = pink;
+    }
```
Comment (estebanmino) [inline src/components/discover-sheet/GasCard.tsx:249]: we could use the new `getColorForGwei`

File: src/design-system/color/palettes.ts [hunk 2]
```diff
@@ -5,7 +5,7 @@ export const colors = {
   appleBlueLight: '#1F87FF',
   black: '#000000',
   blackTint: '#12131a',
-  grey: 'rgb(60, 66, 82)',
+  grey: '#3C4252',
```
Comment (estebanmino) [inline src/design-system/color/palettes.ts:8]: checking if this one was on purpose, we had a change like this in the last profiles PR

File: src/components/discover-sheet/GasCard.tsx [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+                  </Box>
+                </Box>
+              </Inset>
+            </AccentColorProvider>
+          </Box>
+        </AccentColorProvider>
+      </ButtonPressAnimation>
+    </Animated.View>
+  );
+}
```
Comment (estebanmino) [inline src/components/discover-sheet/GasCard.tsx:353]: nice touch

https://user-images.githubusercontent.com/12115171/174107841-cb1ef2b3-c1b5-4c96-9dd5-02bcf66175c0.mov

h

**Review by estebanmino — COMMENTED** [review]
lint and e2e are failing, for `e2e/registerENSFlow` it's failing in the `Should go to ENS flow pressing the ENS banner` step

File: src/design-system/color/palettes.ts (hunk 2 above)
Comment (christianbaroni) [reply src/design-system/color/palettes.ts:8]: was intentional, we use some transforms like `color + '10'` to render a color at 10% opacity, but that only works with hex colors. this was the only solid color in palettes.ts that still used rgb, so `grey + '10'` was breaking

File: src/components/discover-sheet/DiscoverHome.js [hunk 4]
```diff
@@ -1,25 +1,49 @@
...
+                  <ENSSearchCard />
+                </Columns>
+                <ENSCreateProfileCard />
+                <DPICard />
+              </Stack>
+            </Inset>
+          ) : (
+            <Stack space={{ custom: 21 }}>
+              <TopMoversSection />
+              <PulseIndex />
```
[the remark this answers is not in the document]
Comment (christianbaroni) [reply src/components/discover-sheet/DiscoverHome.js:37]: whoops fixed

File: src/components/discover-sheet/DPICard.tsx [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+import Routes from '@rainbow-me/routes';
+import { useTheme } from '@rainbow-me/theme';
+import { ethereumUtils } from '@rainbow-me/utils';
+
+export default function DPICard() {
+  const { nativeCurrency } = useAccountSettings();
+  const { navigate } = useNavigation();
+  const { genericAssets } = useSelector(({ data: { genericAssets } }) => ({
+    genericAssets,
+  }));
```
Comment (terrysahaidak) [inline src/components/discover-sheet/DPICard.tsx:34]: Can we replace it with `useStore` call so we can basically use `store.getState().data.genericAssets` inside the handlePress. In that case, we won't be rerendering this component in the background every time we receive new genericAssets but still be using the latest version of genericAssets in the handler.

File: src/components/discover-sheet/DPICard.tsx [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+              },
+              ios: [
+                {
+                  blur: 24,
+                  color: 'accent',
+                  offset: { x: 0, y: 8 },
+                  opacity: 0.35,
+                },
+              ],
+            },
```
Comment (terrysahaidak) [inline src/components/discover-sheet/DPICard.tsx:87]: can we extract it out of the component?

File: src/components/discover-sheet/DPICard.tsx [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+                    </Column>
+                  </Columns>
+                  <ButtonPressAnimation onPress={handlePress} scaleTo={0.92}>
+                    <Box
+                      style={{
+                        shadowColor: colors.shadowBlack,
+                        shadowOffset: { height: 4, width: 0 },
+                        shadowOpacity: 0.15,
+                        shadowRadius: 6,
+                      }}
```
Comment (terrysahaidak) [inline src/components/discover-sheet/DPICard.tsx:136]: this too

File: src/components/discover-sheet/ENSCreateProfileCard.tsx [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+                  offset: { x: 0, y: 8 },
+                  opacity: colorMode === 'dark' ? 0.3 : 0.1,
+                },
+                {
+                  blur: 6,
+                  offset: { x: 0, y: 2 },
+                  opacity: 0.02,
+                },
+              ],
+            },
```
Comment (terrysahaidak) [inline src/components/discover-sheet/ENSCreateProfileCard.tsx:85]: can we move it just to a variable outside of `return`?

File: src/components/discover-sheet/ENSCreateProfileCard.tsx [hunk 9]
```diff
@@ -0,0 +1,149 @@
...
+            <Box
+              background="accent"
+              borderRadius={24}
+              style={{
+                borderColor: `rgba(0, 0, 0, ${
+                  colorMode === 'dark' ? '0' : '0.1'
+                })`,
+                borderWidth: 0.25,
+                overflow: 'hidden',
+              }}
```
Comment (terrysahaidak) [inline src/components/discover-sheet/ENSCreateProfileCard.tsx:98]: this too

File: src/components/discover-sheet/ENSSearchCard.tsx [hunk 10]
```diff
@@ -0,0 +1,235 @@
...
+              },
+              ios: [
+                {
+                  blur: 24,
+                  color: 'accent',
+                  offset: { x: 0, y: 8 },
+                  opacity: 0.35,
+                },
+              ],
+            },
```
Comment (terrysahaidak) [inline src/components/discover-sheet/ENSSearchCard.tsx:115]: same here - outside the component

File: src/components/discover-sheet/ENSSearchCard.tsx [hunk 11]
```diff
@@ -0,0 +1,235 @@
...
+                            alignHorizontal="center"
+                            alignVertical="center"
+                          >
+                            <AccentColorProvider color="#80D4FF">
+                              <Box background="accent">
+                                <Box
+                                  as={LinearGradient}
+                                  colors={[
+                                    colors.alpha(colors.whiteLabel, 0.8),
+                                    colors.alpha(colors.whiteLabel, 0),
```
Comment (terrysahaidak) [inline src/components/discover-sheet/ENSSearchCard.tsx:199]: would be cool to define these colors in the theme instead

File: src/components/discover-sheet/GasCard.tsx [hunk 12]
```diff
@@ -0,0 +1,353 @@
...
+      stopPollingGasFees();
+    }
+  }, [isFocused, startPollingGasFees, stopPollingGasFees]);
+
+  const { NORMAL } = gasUtils;
+
+  const currentGwei = add(
+    currentBlockParams?.baseFeePerGas?.gwei,
+    gasFeeParamsBySpeed[NORMAL]?.maxPriorityFeePerGas?.gwei
+  );
```
Comment (terrysahaidak) [inline src/components/discover-sheet/GasCard.tsx:82]: useMemo since it uses bignumbers

File: src/components/discover-sheet/GasCard.tsx [hunk 13]
```diff
@@ -0,0 +1,353 @@
...
+    } else if (gwei < 40) {
+      return lang.t('discover.gas.low');
+    } else if (gwei < 100) {
+      return lang.t('discover.gas.average');
+    } else if (gwei < 200) {
+      return lang.t('discover.gas.high');
+    } else {
+      return lang.t('discover.gas.surging');
+    }
+  };
```
Comment (terrysahaidak) [inline src/components/discover-sheet/GasCard.tsx:197]: I guess these can be defined outside the component

Comment (terrysahaidak) [reply src/components/discover-sheet/GasCard.tsx:197]: you can even add `worklet;` and use on both JS and UI thread

File: src/components/discover-sheet/GasCard.tsx [hunk 14]
```diff
@@ -0,0 +1,353 @@
...
+                    blur: 24,
+                    offset: { x: 0, y: 8 },
+                    opacity: colorMode === 'dark' ? 0.3 : 0.1,
+                  },
+                  {
+                    blur: 6,
+                    offset: { x: 0, y: 2 },
+                    opacity: 0.02,
+                  },
+                ],
```
Comment (terrysahaidak) [inline src/components/discover-sheet/GasCard.tsx:294]: same here.
we should definitely figure out a better way of doing custom things

File: src/components/discover-sheet/GasCard.tsx [hunk 15]
```diff
@@ -0,0 +1,353 @@
...
+                  <Stack>
+                    <Stack space={{ custom: 14 }}>
+                      <AnimateNumber
+                        formatter={formatGasPrice}
+                        interval={2}
+                        renderContent={renderGweiText}
+                        timing={(progress: number) => {
+                          var t = progress;
+                          return 1 - --t * t * t * t;
+                        }}
```
Comment (terrysahaidak) [inline src/components/discover-sheet/GasCard.tsx:317]:
```suggestion
                        timing={(t: number) => 1 - --t * t * t * t}
```

**Review by terrysahaidak — CHANGES_REQUESTED** [review]
few comments to improve the code style

**Review by jxom — APPROVED** [review]
🥵

**Review by benisgold — COMMENTED** [review]
the app crashes if you click the DeFi Pulse Index widget when there's no internet connection

edit: nevermind

**Review by benisgold — APPROVED** [review]

**Review by terrysahaidak — APPROVED** [review]
I'm just not sure about 300kb image used for the card, could we add multiple sizes at least? we don't really need to render 1300x600px image for every phone :)

**Review by osdnk — APPROVED** [review]

**Comment by BrodyHughes** [comment]
- Trying to do a transaction to change ENS info / buy an ENS has some weird behavior
- on the "create your ENS profile" modal popup - alisha.eth has a blank photo
- everything else you mentioned aside from "Verify that the "register a .eth name" updates when you have pending registrations" works fine. I haven't been able to test this yet due to the weird behavior i mentioned. 
Screen Shot 2022-07-12 at 12 53 40 PM
Screen Shot 2022-07-12 at 1 01 57 PM

**Comment by BrodyHughes** [comment]
Rechecking this now

**Review by BrodyHughes — APPROVED** [review]
Looks good. Everything functioning as intended. Theres an oddity with the gas ticking up 1 cent at a time, but I can't get it to reproduce consistently.

**Pull request merged**